hummingbot/connector/exchange/huobi/huobi_exchange.py

```diff
@@ -0,0 +1,423 @@
+
+    async def _process_trade_event(self, trade_event: Dict[str, Any]):
+        client_order_id = trade_event["clientOrderId"]
+        tracked_order = self.in_flight_orders.get(client_order_id, None)
+
+        if tracked_order:
+            fee = TradeFeeBase.new_spot_fee(
+                fee_schema=self.trade_fee_schema(),
+                trade_type=tracked_order.trade_type,
+                percent_token=trade_event["feeCurrency"].upper(),
+                flat_fees=[TokenAmount(amount=Decimal(trade_event["transactFee"]), token=trade_event["feeCurrency"].upper())]
+            )
+            trade_update = TradeUpdate(
+                trade_id=str(trade_event["tradeId"]),
+                client_order_id=client_order_id,
+                exchange_order_id=str(trade_event["orderId"]),
+                trading_pair=tracked_order.trading_pair,
+                fee=fee,
+                fill_base_amount=Decimal(trade_event["tradeVolume"]),
+                fill_quote_amount=Decimal(trade_event["tradeVolume"]) * Decimal(trade_event["tradePrice"]),
+                fill_price=Decimal(trade_event["tradePrice"]),
+                fill_timestamp=trade_event["tradeTime"] * 1e-3,
+            )
+            self._order_tracker.process_trade_update(trade_update)
+
+    async def _update_trading_fees(self):
+        pass
+
+    def supported_order_types(self):
+        return [OrderType.LIMIT, OrderType.LIMIT_MAKER]
+
+    async def _place_order(self,
+                           order_id: str,
+                           trading_pair: str,
+                           amount: Decimal,
+                           trade_type: TradeType,
+                           order_type: OrderType,
+                           price: Decimal):
+        path_url = CONSTANTS.PLACE_ORDER_URL
+        side = trade_type.name.lower()
+        order_type_str = "limit" if order_type is OrderType.LIMIT else "limit-maker"
+        if not self._account_id:
+            await self._update_account_id()
+        params = {
+            "account-id": self._account_id,
+            "amount": f"{amount}",
+            "client-order-id": order_id,
+            "symbol": web_utils.convert_to_exchange_trading_pair(trading_pair),
+            "type": f"{side}-{order_type_str}",
+        }
+        if order_type is OrderType.LIMIT or order_type is OrderType.LIMIT_MAKER:
+            params["price"] = f"{price}"
+        exchange_order_id = await self._api_post(
+            path_url=path_url,
+            params=params,
+            data=params,
+            is_auth_required=True
+        )
+        return str(exchange_order_id["data"]), self.current_timestamp
+
+    async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
+        if tracked_order is None:
+            raise ValueError(f"Failed to cancel order - {order_id}. Order not found.")
+        path_url = CONSTANTS.CANCEL_ORDER_URL.format(tracked_order.exchange_order_id)
```

Comment:
We should use the endpoint that cancels orders based on their client order id, and not the one for exchange order id.